### PR TITLE
feat: :sparkles: Ambito API connected.

### PR DESCRIPTION
**Pull request by AndresEstebanGonzalez**
Ambito API is now connected and show the dolar blue value. The value automatically refreshes every 5 minutes.

## Pull request type

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## If its a bug, what was the behavior?

-

Card Link: https://trello.com/c/pB0B2s3J

## What is the new behavior?

App is now able to fetch the dolar value using the Ambito API.

## Other information

-

## Preview

Paste your screenshots here! :)

### CODE REVIEW

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -7,6 +8,28 @@ import Typography from '@mui/material/Typography';
...
+  const fetchDolarValue = async () => {
+    const response = await fetch('https://mercados.ambito.com/home/general3');
+    return response.json();
+  };
+
+  const { data, status } = useQuery({
+    queryKey: ['dolarValue'],
+    queryFn: fetchDolarValue,
+    refetchInterval: DOLAR_REFETCH_INTERVAL,
+  });
```
Comment (nathan-hyan) [inline src/components/Navbar.tsx:21]: Move this logic to a custom hook.

File: src/components/Navbar.tsx [hunk 2]
```diff
@@ -7,6 +8,28 @@ import Typography from '@mui/material/Typography';
 import PriceDisplay from './PriceDisplay';
 
 export default function Navbar() {
+  const DOLAR_REFETCH_INTERVAL = 300000;
```
Comment (nathan-hyan) [inline src/components/Navbar.tsx:11]: Move this constant to constants.ts

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -7,6 +8,28 @@ import Typography from '@mui/material/Typography';
...
+  const fetchDolarValue = async () => {
+    const response = await fetch('https://mercados.ambito.com/home/general3');
```
Comment (nathan-hyan) [inline src/components/Navbar.tsx:13]: Don't ever hardcode api urls. Move the url to an .env file

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -7,6 +8,28 @@ import Typography from '@mui/material/Typography';
...
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
+  }
+
+  if (status === 'error') {
+    return <div>Error: Failed to fetch Dolar value.</div>;
+  }
+
+  const DOLAR_INFORMAL = data[1].venta;
```
Comment (nathan-hyan) [inline src/components/Navbar.tsx:31]: Export this value from the custom hook itself

**Review by nathan-hyan — COMMENTED** [review]
Amazing work, some comments!

File: .env [hunk 5]
```diff
@@ -0,0 +1 @@
+VITE_AMBITO_URL=https://mercados.ambito.com/home/general3
```
Comment (nathan-hyan) [inline .env:1]: Add .env to .gitignore.

*NEVER* push .env to a repository since here's where all your secrets will be added

**Review by nathan-hyan — CHANGES_REQUESTED** [review]
There's a critical error in this pr, fix ASAP

Also, before you push the fix for the error, add proper types for the response of the Ambito API as specified in the next card: https://trello.com/c/pB0B2s3J

```
{
  "compra": string;
  "venta": string;
  "fecha": string;
  "class-variacion": "up" | "down" | "up-rp"
}
```

File: src/constants/constants.ts
Comment (nathan-hyan) [inline src/constants/constants.ts:1]: This constant can be inferred, no need for typing it

File: src/hooks/useFetchDolarValue.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    return response.json();
+  };
+  const { data, status } = useQuery<Data, Error>({
+    queryKey: ['dolarValue'],
+    queryFn: fetchDolarValue,
+    refetchInterval: DOLAR_REFETCH_INTERVAL_MS,
+  });
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
```
Comment (nathan-hyan) [inline src/hooks/useFetchDolarValue.tsx:25]: Dont return JSX in a hook.

You can do

```
let output: string

if (status === 'loading') {
    output = 'loading...'
}

if (status === 'error') {
    output = 'There was an error'
}

output = data?.[1]?.venta || 'There was an error'

return output
```

File: .env (hunk 5 above)
Comment (nathan-hyan) [reply .env:1]: This file should not exist.

**Review by nathan-hyan — COMMENTED** [review]
Please check deployment messages

**Review by nathan-hyan — COMMENTED** [review]
Please check comments and fix asap

File: src/hooks/useFetchDolarValue.tsx (hunk 6 above)
Comment (AndresEstebanGonzalez) [reply src/hooks/useFetchDolarValue.tsx:25]: That looks so much more neater! Thanks!

**Review by nathan-hyan — APPROVED** [review]

**Pull request merged**